lib/ABI/FunctionType/ValueDistributor.h

```diff
@@ -20,16 +20,30 @@ struct DistributedValue {
   /// A list of registers the argument uses.
   RegisterVector Registers = {};
 
-  /// The total size of the argument (including padding if necessary) in bytes.
+  /// The total size of the argument in bytes WITHOUT any padding.
   uint64_t Size = 0;
 
+  /// The total size of padding that has to be added to the previous argument
+  /// because of alignment specifics.
+  uint64_t PrePaddingSize = 0;
+
+  /// The total size of padding that has to be added to the current argument.
+  uint64_t PostPaddingSize = 0;
```

Comment:
These fields are set but never read. What's their purpose?

Reply:
OK, can you add a comment stating that as of now these are mainly for debugging purposes?
If this change, we can drop it.

Reply:
Done

I also used that chance to update the log, please take a look at the new `ValueDistributor: greatly improve success log message` commit.